malariagen_data/ag3.py

```diff
@@ -2489,6 +2490,92 @@ def aa_allele_frequencies(
 
         return df_aaf
 
+    def plot_frequencies_heatmap(
+        self, df, index, index_name, max_len=100, width=500, colorbar=True
+    ):
+
+        """Generate a Plotly.express frequency heatmap using pandas DataFrame output from ag3.snp_allele_frequencies()
+        or ag3.gene_cnv_frequencies() that has been user-filtered to just rows of interest (fewer rows than max_len).
+
+        Parameters
+        ----------
+        df : pandas DataFrame
+           Filtered output from ag3.snp_allele_frequencies() or ag3.gene_cnv_frequencies.
+        index : list of str
+            A list of one or more column headers present in the input dataframe, these become
+            the heatmap y-axis row labels. When taken together, these columns must produce a unique index.
+        index_name : str
```

Comment:
It would be good to make this optional. I.e., the user doesn't have to specify an `index_name`, in which case no Y axis label appears.